lib/containers/runtime.pm

```diff
@@ -131,18 +222,178 @@ sub remove_image {
+        assert_equals(0, scalar @{$self->enum_images()},     "images have not been removed");
+    }
+}
+
+# Container runtime subclasses here
+package containers::runtime::docker;
+use Mojo::Base 'containers::runtime';
+use testapi;
+use containers::utils qw(registry_url);
+use utils qw(systemctl file_content_replace);
+has runtime => 'docker';
+
+sub setup_registry {
     my ($self) = shift;
-    # copy from common > clean_container_host
-    $self->_rt_assert_script_run("stop \$($self->{runtime} ps -q)", 180) if script_output("$self->{runtime} ps -q | wc -l") != '0';
-    $self->_rt_assert_script_run("system prune -a -f",              180);
-    assert_equals(0, scalar @{$self->enum_containers()}, "containers have not been removed");
-    assert_equals(0, scalar @{$self->enum_images()},     "images have not been removed");
+    my $registry = registry_url();
+    # Allow our internal 'insecure' registry
+    assert_script_run(
+        'echo "{ \"debug\": true, \"insecure-registries\" : [\"localhost:5000\", \"registry.suse.de\", \"' . $registry . '\"] }" > /etc/docker/daemon.json');
+    assert_script_run('cat /etc/docker/daemon.json');
+    systemctl('restart docker');
+    record_info "setup $self->runtime", "deamon.json ready";
+}
+
+sub up {
+    my ($self, $image_name, %args) = @_;
+    die 'image name or id is required' unless $image_name;
+    my $mode           = $args{daemon}         ? '-d'                 : '-i';
+    my $remote         = $args{cmd}            ? "$args{cmd}"         : '';
+    my $name           = $args{name}           ? "--name $args{name}" : '';
+    my $keep_container = $args{keep_container} ? ''                   : '--rm';
+    my $params         = sprintf qq(%s %s %s), $keep_container, $mode, $name;
+    my $ret            = $self->_rt_script_run(sprintf qq(run %s %s %s), $params, $image_name, $remote, timeout => $args{timeout});
+    record_info "Remote run on $image_name", "options $keep_container $mode $name $image_name $remote";
+    return $ret;
+}
+
+package containers::runtime::podman;
+use Mojo::Base 'containers::runtime';
+use testapi;
+use containers::utils qw(registry_url);
+use utils qw(systemctl file_content_replace);
+has runtime => "podman";
+
+sub setup_registry {
+    my ($self) = shift;
+    my $registry = registry_url();
+
+    assert_script_run "curl " . data_url('containers/registries.conf') . " -o /etc/containers/registries.conf";
+    assert_script_run "chmod 644 /etc/containers/registries.conf";
+    file_content_replace("/etc/containers/registries.conf", REGISTRY => $registry);
+}
+
+
+package containers::runtime::buildah;
+use Mojo::Base 'containers::runtime';
+use testapi;
+use containers::utils qw(registry_url);
+use utils qw(systemctl file_content_replace);
+has runtime => "buildah";
+
+sub cleanup_system_host {
+    my ($self, $assert) = @_;
+    $assert // 1;
+    $self->_rt_assert_script_run("rm --all");
```

Comment:
I would implement this in a way that we call the super class function passing the cmd (or array of cmds) to clean the host. 

```
my $cmd = ('rm --all', 'rmi --all --force');
$self->SUPER::cleanup_system_host($cmd);
```

And in the super class, you leave the cmds that are already there by default.

Reply:
let me check but in first glance it makes sense

Reply:
@jlausuch i would like to leave it as it is for now. The reasons are:
- seems a bit over-engineering for those two lines
- the overridden subroutine uses xargs
- passing cmd makes the function less specific. in theory you could execute whatever

Reply:
In my opinion Jose's suggestion makes still sense over there.